File: 101/02 贪心/01_455.cc
class Solution {
  public:
    int findContentChildren(vector<int>& children, vector<int>& cookies) {
      sort(children.begin(), children.end());
      sort(cookies.begin(), cookies.end());
      int i_children = 0;
      int i_cookies = 0;
      while (i_children < children.size() && i_cookies < cookies.size()) {
        if (children[i_children] <= cookies[i_cookies]) ++i_children;
        ++i_cookies;
      }
      return i_children;
    }
};

File: 101/02 贪心/01_455.js
/**
 * @param {number[]} children
 * @param {number[]} cookies
 * @return {number}
 */
function findContentChildren(children, cookies) {
  const compareNum = (a, b) => a - b;
  children.sort(compareNum);
  cookies.sort(compareNum);
  let i_children = 0;
  let i_cookies = 0;
  while (i_children < children.length && i_cookies < cookies.length) {
    if (children[i_children] <= cookies[i_cookies]) ++i_children;
    ++i_cookies;
  }
  return i_children;
}
